File: docs/snippets/all/howto/load_mcap.py
"""Load an MCAP file using the Python SDK."""

import sys

import rerun as rr

path_to_mcap = sys.argv[1]

# Initialize the SDK and give our recording a unique name
rr.init("rerun_example_load_mcap", spawn=True)

# Load the MCAP file
rr.log_file_from_path(path_to_mcap)

File: docs/snippets/all/howto/load_mcap.cpp
#include <cstdio>
#include <rerun.hpp>

int main(int argc, char* argv[]) {
    if (argc != 2) {
        printf("Usage: %s <path_to_mcap>", argv[0]);
        return 1;
    }

    const std::string path_to_mcap = argv[1];

    // Initialize the SDK and give our recording a unique name
    const auto rec = rerun::RecordingStream("rerun_example_load_mcap");
    rec.spawn().exit_on_failure();

    // Load the MCAP file
    rec.log_file_from_path(path_to_mcap);

    return 0;
}
